apps/extension/src/shared/components/numeraires-form.tsx

```diff
@@ -13,39 +12,36 @@ const useNumerairesSelector = (state: AllSlices) => {
     selectedNumeraires: state.numeraires.selectedNumeraires,
     selectNumeraire: state.numeraires.selectNumeraire,
     saveNumeraires: state.numeraires.saveNumeraires,
-    networkChainId: state.network.chainId,
   };
 };
 
 export const NumeraireForm = ({
   isOnboarding,
   onSuccess,
+  chainId,
 }: {
+  chainId?: string;
   isOnboarding?: boolean;
-  onSuccess: () => void | Promise<void>;
+  onSuccess: () => void;
 }) => {
-  const { chainId } = useChainIdQuery();
-  const { selectedNumeraires, selectNumeraire, saveNumeraires, networkChainId } =
-    useStore(useNumerairesSelector);
-
-  // 'chainId' from 'useChainIdQuery' is not available during onboarding,
-  // this forces you to use two sources to guarantee 'chainId' for both settings and onboarding
-  const { numeraires } = useNumeraires(chainId ?? networkChainId);
+  const { selectedNumeraires, selectNumeraire, saveNumeraires } = useStore(useNumerairesSelector);
+  const { numeraires, isLoading, isError } = useNumeraires(chainId);
 
+  // If query errors or there aren't numeraires in the registry, can skip
   useEffect(() => {
-    if (numeraires.length === 0) {
-      void onSuccess();
+    if (isError || (!isLoading && numeraires.length === 0)) {
+      onSuccess();
     }
-  }, [numeraires]);
+  }, [numeraires.length, isError, isLoading]);
 
   const [loading, setLoading] = useState(false);
 
   const handleSubmit = () => {
     setLoading(true);
     void (async function () {
       await saveNumeraires();
-      await chrome.runtime.sendMessage(ServicesMessage.ChangeNumeraires);
-      await onSuccess();
+      void chrome.runtime.sendMessage(ServicesMessage.ChangeNumeraires);
```

Comment:
Does not feel necessary to await the response here, hold up the UI from progressing. The fire-and-forget works.